src/testFixtures/java/uk/gov/di/ipv/cri/common/library/stepdefinitions/WellKnownJwksSteps.java

```diff
@@ -106,4 +158,89 @@ public void at_least_one_key_use_is_for(String keyUse, String keyType)
+    }
+
+    @SuppressWarnings("unchecked")
+    @Then("that request is verified by the CRI")
+    public void theRequestIsVerified() throws ParseException, ClientConfigurationException {
+        verifyRequest(decryptedJwt);
+
+        JWSHeader header = decryptedJwt.getHeader();
+
+        assertThat(header.getAlgorithm().getName(), is("ES256"));
+        assertThat(header.getType().getType(), is("JWT"));
+        assertThat(header.getKeyID(), notNullValue());
+
+        JWTClaimsSet claimsSet = decryptedJwt.getJWTClaimsSet();
+        var claims = claimsSet.getClaims();
+
+        var sharedClaims = (Map<String, Object>) claims.get("shared_claims");
+        var names = (List<Map<String, Object>>) sharedClaims.get("name");
+        var nameParts = (List<Map<String, Object>>) names.get(0).get("nameParts");
+        var birthDates = (List<Map<String, Object>>) sharedClaims.get("birthDate");
+        var addresses = (List<Map<String, Object>>) sharedClaims.get("address");
+
+        assertThat(sharedClaims, allOf(hasKey("name"), hasKey("birthDate"), hasKey("address")));
+
+        assertThat(names, everyItem(hasKey("nameParts")));
+        assertThat(nameParts, everyItem(allOf(hasKey("type"), hasKey("value"))));
+        assertThat(birthDates, everyItem(hasKey("value")));
+
+        assertThat(
+                addresses,
+                everyItem(
+                        allOf(
+                                hasKey("addressLocality"),
+                                hasKey("buildingNumber"),
+                                hasKey("postalCode"),
+                                hasKey("streetName"),
+                                hasKey("validFrom"))));
+    }
+
+    private void verifyRequest(SignedJWT signedJWT) throws ClientConfigurationException {
+        JWTVerifier jwtVerifier = new JWTVerifier();
+
+        jwtVerifier.verifyAuthorizationJWT(getSSMClientConfig(), signedJWT);
+    }
+
+    private URI getPublicEncryptionJwkUri() {
+        return new URIBuilder(publicKeyJwksBasePath)
+                .setPath(this.httpClientHelper.createUriPath(".well-known/jwks.json"))
+                .build();
+    }
+
+    private KmsClient createKmsClient() {
+        return KmsClient.builder()
+                .region(Region.EU_WEST_2)
+                .credentialsProvider(DefaultCredentialsProvider.create())
+                .build();
+    }
+
+    private SsmClient createSsmClient() {
+        return SsmClient.builder()
+                .region(Region.EU_WEST_2)
+                .credentialsProvider(DefaultCredentialsProvider.create())
+                .build();
+    }
+
+    private Map<String, String> getSSMClientConfig() {
+        return Arrays.stream(CriStubClientEnum.values())
+                .collect(toMap(CriStubClientEnum::getConfigName, stubClientConfig::getValue));
+    }
+
+    private String getParameter(String path) {
+        return ssmHelper.getParameterValueByName(path);
+    }
```

Comment:
I wonder if methods like these could live in some sort of test helpers file?

Reply:
I had my own implementation of `getParameter` when I rebased on Mariese PR, i decided to re-use `ssmHelper.getParameterValueByName(path)`  I have removed `getParameter`  and just used `ssmHelper.getParameterValueByName(path)`  I thought the code would sprawl further here

```
        String authEncryptionKeyId =
                ssmHelper.getParameterValueByName(
                        format(
                                "/%s/%s",
                                getEnvOrDefault("COMMON_STACK_NAME", "common-cri-api"),
                                "AuthRequestKmsEncryptionKeyId"));
```
but it's not too bad